### ISSUE TITLE
Discuss client-server model

### ISSUE THREAD

**Issue by JosephSalisbury** [giantswarm/inago#237]
In https://github.com/giantswarm/inago/issues/215 we noted the possibility of requiring some kind of  server.

This issue is present to pull together what we can't do without a server, and whether we want to move in that direction.

Some items noted already that would require a server:
- Maintaining a dynamic number of units on a server (fleet-unit-replicator style)
- Lock actions on resources to e.g. prevent concurrent updates of the same slices
- Get slice identification right by using a centralized store
- Viewing currently occurring actions

**Comment by kopiczko** [comment]
Correct me if I'm wrong but I think all of that can be achieved with use of etcd, no server required. Just my 2c.

**Comment by zeisss** [comment]
@kopiczko Yes, this can be achieved by giving the inago user access to etcd to ensure there is a lock. But this adds additional complexity to a client process, which especially given it is run on developer machines has the tendency to just be killed / put to sleep e.g. for lunch.

A server would bring the benefit of protecting the rollout from bad client behavior.
